src/Nfts/ImageGallery/ImageGallery.tsx

```diff
@@ -58,81 +88,159 @@ interface ModeratedImageProps {
 
 const ModeratedImage = ({fingerprint, image, text, onPress}: ModeratedImageProps) => {
   const wallet = useSelectedWallet()
-  const moderationStatusQuery = useQuery({
-    queryKey: [wallet.id, 'nft', fingerprint],
-    queryFn: () => wallet.fetchNftModerationStatus(fingerprint),
-  })
+  const moderationStatusQuery = useNftModerationStatus({wallet, fingerprint})
 
   useEffect(() => {
     if (moderationStatusQuery.data === 'pending') {
       moderationStatusQuery.refetch()
     }
   }, [moderationStatusQuery.data, moderationStatusQuery])
 
-  const showSkeleton = moderationStatusQuery.isLoading
-  const isImageApproved = moderationStatusQuery.data === 'green'
+  const isPendingReview = moderationStatusQuery.data === 'pending' || moderationStatusQuery.data === 'manual_review'
+  const showSkeleton = moderationStatusQuery.isLoading || isPendingReview
+
+  const isImageApproved = moderationStatusQuery.data === 'approved'
+  const isImageWithConsent = moderationStatusQuery.data === 'consent'
+  const isImageBlocked = moderationStatusQuery.data === 'blocked'
 
   if (showSkeleton) {
     return <SkeletonImagePlaceholder />
   }
 
   return (
-    <TouchableOpacity onPress={onPress} style={styles.imageContainer}>
+    <TouchableOpacity disabled={isImageBlocked} onPress={onPress} style={[styles.imageContainer]}>
       {isImageApproved ? (
-        <>
-          <Image source={{uri: image}} style={styles.image} />
-          <Spacer height={8} />
-          <Text style={styles.textTop}>{text}</Text>
-          <Spacer height={13} />
-        </>
-      ) : (
-        <>
-          <View style={styles.image} />
-          <Spacer height={8} />
-          <Text style={styles.textTop}>Image is not approved</Text>
-          <Spacer height={13} />
-        </>
-      )}
+        <ApprovedNFT text={text} uri={image} />
+      ) : isImageWithConsent ? (
+        <RequiresConsentNFT text={text} uri={image} />
+      ) : isImageBlocked ? (
+        <BlockedNFT text={text} />
+      ) : null}
     </TouchableOpacity>
   )
 }
 
-function SkeletonImagePlaceholder() {
+function BlockedNFT({text}: {text: string}) {
+  return (
+    <>
```

Comment:
why use a fragment? why not View?

Reply:
I've been asked to use a Fragment instead of a View earlier when not needed, to not add an extra View element

Reply:
fragments break encapsulation
I only use them when bundling modals